### PR TITLE
feat: This adds sample tests based on the new GO test framework

### PR DESCRIPTION
**Pull request by g-awmalik**
- adds two tests based on the GO test framework
   test/integration/discover_test.go
   test/integration/mysql-public/mysql_public_test.go

- updates the dev-tools docker version
- fixes headers on two yml files

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_headers
 All files need a license header. Please make sure all your files include the appropriate header. A helper tool available [here](https://github.com/google/addlicense). 
 ```bash 
Checking file headers
2 files have incorrect boilerplate headers:
test/integration/discover_test.go
test/integration/mysql-public/mysql_public_test.go
 ``` 
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
test/fixtures/mysql-public/outputs.tf
--- old/test/fixtures/mysql-public/outputs.tf
+++ new/test/fixtures/mysql-public/outputs.tf
@@ -32,7 +32,7 @@
 output "mysql_user_pass" {
   value       = module.example.mysql_user_pass
   description = "The password for the default user. If not set, a random one will be generated and available in the generated_user_password output variable."
-  sensitive = true
+  sensitive   = true
 }
 
 output "public_ip_address" {
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

File: Makefile [hunk 1]
```diff
@@ -18,7 +18,7 @@
 # Make will use bash instead of sh
 SHELL := /usr/bin/env bash
 
-DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 0.13
+DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 1.0
```
Comment (bharathkkb) [inline Makefile:21]: Lets update readme like https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/965/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L15-R299

File: test/integration/discover_test.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+package test
+
+import (
+	"testing" // should be imported to enable testing for GO modules
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/tft" // should be
+	// imported to enable the Blueprints helper modules for the test framework
+)
+
+func TestAll(t *testing.T) { // entry function for the test; can be named as Test*
```
Comment (bharathkkb) [inline test/integration/discover_test.go:24]: Where are we invoking the test in CI?
I suggest we add a new build stage for go tests before we phase out the others

File: test/integration/go.mod [hunk 3]
```diff
@@ -0,0 +1,33 @@
+module github.com/GoogleCloudPlatform/cloud-foundation-toolkit
```
Comment (bharathkkb) [inline test/integration/go.mod:1]: nit: 
```suggestion
module github.com/terraform-google-modules/terraform-google-sql-db/test/integration
```

File: test/integration/mysql-public/mysql_public_test.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	const gceZone = `us-central1-c`
+	const region = `us-central1`
+	const tier = `db-n1-standard-1`
+	const authNetName = `sample-gcp-health-checkers-range`
+	const authNetCidr = `130.211.0.0/28`
+	const dbFlagName = `log_bin_trust_function_creators`
+	const dbFlagValue = `on`
+
+	mySqlT := tft.NewTFBlueprintTest(t) // initialize Terraform test from the Blueprints test framework
+	//bpt.TestBlueprint(t, mySqlT,
```
Comment (bharathkkb) [inline test/integration/mysql-public/mysql_public_test.go:41]: nit: old code

File: test/integration/mysql-public/mysql_public_test.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mysql_public
+
+import (
+	// import standard formatting and testing packages
```
Comment (bharathkkb) [inline test/integration/mysql-public/mysql_public_test.go:18]: nit: I think we can remove this since its from stdlib

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks @g-awmalik

**Comment by g-awmalik** [comment]
Closing this PR based on my discussion with @bharathkkb to isolate the blueprints fixes from the test framework addition.

**Pull request closed without merging**